src/client/pythonEnvironments/common/commonUtils.ts

```
@@ -22,79 +25,173 @@ type FileFilterFunc = (filename: string) => boolean;
  * @param recurseLevels : Number of levels to search for from the root directory.
  * @param filter : Callback that identifies directories to ignore.
  */
-export function findInterpretersInDir(
+export async function* findInterpretersInDir(
     root: string,
     recurseLevel?: number,
     filterSubDir?: FileFilterFunc,
-    ignoreErrors?: boolean,
+    ignoreErrors = false,
```

I think we should be ignoring errors and continuing by default, rather than failing the whole iteration.

---

I'm only going with the current behavior.  We can switch to ignore-by-default in a separate PR if that's what we want to do.

---

Please do open a work item in that case.

---

Normally the person who has a concern with the status quo takes charge of driving any change. 😄

---

Alright. 
It's just that I had also mentioned this in the original PR you authored where this option was introduced: https://github.com/microsoft/vscode-python/pull/14994#discussion_r553032360. It was marked as resolved but apparently wasn't.